.storybook/stories/DataFiltering/TableFilteringMedium.tsx

```diff
@@ -30,15 +47,64 @@ export const TableFilteringMedium = ({
 }: {
 	selectable?: boolean;
 }) => {
+	const { updateData } = useDataContext();
+
+	const sectionAlertRef = useRef(null);
 	const [isOpen, toggleIsOpen] = useToggleState(false, true);
 
+	const [currentItem, setCurrentItem] = useState<BusinessForAudit>();
+	const [showSuccessMessage, setShowSuccessMessage] = useState(false);
+
+	const formRef = useRef(null);
+	const [isDrawerOpen, openDrawer, closeDrawer] = useTernaryState(false);
+	const onSubmitForm: MouseEventHandler<HTMLButtonElement> = (event) => {
+		event.preventDefault();
+
+		if (!currentItem) return;
+		updateData?.({
+			...currentItem,
+			...(radioStatusValue && { status: radioStatusValue }),
+		});
+
+		setShowSuccessMessage(true);
+
+		closeDrawer();
+	};
+	const onOpenDrawer = (newCurrentItem?: BusinessForAudit) => {
+		openDrawer();
+		setShowSuccessMessage(false);
+		setCurrentItem(newCurrentItem);
+		setRadioStatusValue(newCurrentItem?.status);
+	};
+
+	const [radioStatusValue, setRadioStatusValue] =
+		useState<BusinessForAuditStatus>();
+	const handlerForKey = useCallback(
+		(key: BusinessForAuditStatus) => () => setRadioStatusValue(key),
+		[]
+	);
+	const isChecked = (key: BusinessForAuditStatus) => key === radioStatusValue;
+
 	// IDs for accordion to ensure accessibility
 	const buttonId = 'filter-button';
 	const bodyId = 'filter-body';
 
 	return (
 		<Stack gap={1.5}>
 			<H2 id={headingId}>Audits</H2>
+
+			<Box css={showSuccessMessage ? undefined : { display: 'none' }}>
+				<SectionAlert
+					id="section-alert"
+					ref={sectionAlertRef}
```

Comment:
You don't appear to be doing anything with this ref?

Reply:
Yeah, I left it in here in case we need to programatically focus it. Same with `tabIndex` and `id`